text/0000-deprecate-ember-assign.md

```diff
@@ -0,0 +1,52 @@
+---
+Stage: Accepted
+Start Date: 2021-05-26
+Release Date: Unreleased
+Release Versions:
+  ember-source: vX.Y.Z
+  ember-data: vX.Y.Z
+Relevant Team(s): Ember.js
+RFC PR: https://github.com/emberjs/rfcs/pull/750
+---
+
+<!---
+Directions for above:
+
+Stage: Leave as is
+Start Date: Fill in with today's date, YYYY-MM-DD
+Release Date: Leave as is
+Release Versions: Leave as is
+Relevant Team(s): Fill this in with the [team(s)](README.md#relevant-teams) to which this RFC applies
+RFC PR: Fill this in with the URL for the Proposal RFC PR
+-->
+
+# Deprecate Ember.assign
+
+## Summary
+
+Deprecate `Ember.assign` starting in v4.0. Now that Ember is dropping support for IE11, we no longer need `Ember.assign` as a polyfill since `Object.assign`
+is available in all browsers that Ember v4.0+ supports ([CanIUse](https://caniuse.com/mdn-javascript_builtins_object_assign), [MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/assign#browser_compatibility)).
+
+## Motivation
+
+The polyfill is no longer necessary and is a small amount of code that can be removed and remove another Emberism. Apps and addons can use `Object.assign` or object destructring depending on their browser support targets.
+
+## Transition Path
```

Comment:
I think it would be good to detail the path forward by a few categories:

* Libraries - If you still support IE11 (e.g. existing Ember 3.0 usage), you probably should do `const assign = Object.assign || Ember.assign;`
* Apps - Once you've dropped IE11 support, just update `Ember.assign` directly to `Object.assign`

Reply:
I'm proposing to only add this deprecation to Ember 4.0, so consuming apps will already have dropped IE11 support and Ember 3.x users won't see it.

Should I change that and assume the deprecation will land in Ember 3.x?

Reply:
It seems very common that an addon may support apps that are using Ember 3.x and 4.x at the same time, or that the addon isn't ready to do a breaking change release (to drop IE11 support) just yet.

Reply:
Makes sense. Added some IE11/Ember 3.x verbage